packages/asset-swapper/src/utils/quote_report_generator.ts

```diff
@@ -306,16 +306,8 @@ export function dexSampleToReportSource(ds: DexSample, marketOperation: MarketOp
  * Checks if a DEX sample is the one that represents the whole amount requested by taker
  * NOTE: this is used for the QuoteReport to filter samples
  */
-function isDexSampleForTotalAmount(ds: DexSample, marketOperation: MarketOperation, amount: BigNumber): boolean {
-    // input and output map to different values
-    // based on the market operation
-    if (marketOperation === MarketOperation.Buy) {
-        return ds.input === amount;
-    } else if (marketOperation === MarketOperation.Sell) {
-        return ds.output === amount;
-    } else {
-        throw new Error(`Unexpected marketOperation ${marketOperation}`);
-    }
+function isDexSampleForTotalAmount(ds: DexSample, amount: BigNumber): boolean {
+    return ds.input === amount;
```

Comment:
I think I'd suggest this rather than the `===` operator.
```suggestion
    return ds.input.eq(amount);
```